src/vss_tools/exporters/jsonschema.py

```diff
@@ -189,39 +106,90 @@ def cli(
-        no_additional_properties,
-        require_all_properties,
-    )
+    schema = {"$schema": "https://json-schema.org/draft/2020-12/schema", "title": tree.name}
+
+    add_node(schema, tree, datatype_tree, no_additional_properties, require_all_properties, extend_all_attributes)
 
-    # Add data types to the schema
-    if datatype_tree is not None:
-        data_types_json_schema: Dict[str, Any] = {}
-        export_node(
-            data_types_json_schema,
-            datatype_tree,
-            extend_all_attributes,
-            no_additional_properties,
-            require_all_properties,
-        )
-        if extend_all_attributes:
-            signals_json_schema["x-ComplexDataTypes"] = data_types_json_schema
-
-    # VSS models only have one root, so there should only be one
-    # key in the dict
-    assert len(signals_json_schema.keys()) == 1
-    top_node_name = list(signals_json_schema.keys())[0]
-    signals_json_schema = signals_json_schema.pop(top_node_name)
-
-    json_schema = {
-        "$schema": "https://json-schema.org/draft/2020-12/schema",
-        "title": top_node_name,
-        "type": "object",
-        **signals_json_schema,
-    }
     with open(output, "w", encoding="utf-8") as output_file:
-        json.dump(json_schema, output_file, indent=indent, sort_keys=False)
+        json.dump(schema, output_file, indent=indent, sort_keys=False)
+
+
+def find_type_node(datatype_tree: VSSNode | None, fqn: str) -> VSSNode | None:
+    if not datatype_tree:
+        return None
+    return datatype_tree.get_node_with_fqn(fqn)
+
+
+def add_x_attributes(schema: dict[str, Any], node: VSSNode) -> None:
+    data = node.get_vss_data()
+    schema["x-VSStype"] = data.type.value
+    if isinstance(node.data, VSSDataDatatype):
+        schema["x-datatype"] = node.data.datatype
+        if node.data.unit:
+            schema["x-unit"] = node.data.unit
+    if data.deprecation:
+        schema["x-deprecation"] = data.deprecation
+    if isinstance(node.data, VSSDataBranch):
+        schema["x-aggregate"] = node.data.aggregate
+    if data.comment:
+        schema["x-comment"] = data.comment
+
+
+def add_node(
+    schema: dict[str, Any],
+    node: VSSNode,
+    dtree: VSSNode | None,
+    no_additional_props: bool,
+    require_all_properties: bool,
+    extend_all_attributes: bool,
+) -> None:
+    schema["type"] = "object"
+    schema["description"] = node.get_vss_data().description
+    if extend_all_attributes:
+        add_x_attributes(schema, node)
+    if isinstance(node.data, VSSDataDatatype):
+        ref = schema
+        if is_array(node.data.datatype):
+            schema["type"] = "array"
+            schema["items"] = {}
+            ref = schema["items"]
+        datatype = node.data.datatype.rstrip("[]")
+        if datatype in type_map:
+            target_type = type_map[datatype]
+            target_type = type_map[datatype]
+            ref["type"] = target_type[0]
+            if len(target_type) > 1:
```

Comment:
As of today we expect "len" to be either 1 or 3, right? I.e. never anything else.

Reply:
Right now it only will be 1 or 3. Yeah. Thought about specifying at least the lower boundary for uint64. Then we would have 2 :)